src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -87,12 +86,14 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
 
     default String getNodeSpecificTmpFolderPath(FilePath workspace) throws IOException, InterruptedException {
         Computer cmp = workspace.toComputer();
+        if (cmp == null) {
+            throw new IOException(Message.getValue("build.workspace.computer.not.found"));

```

Comment:
In the unlikely case that cmp is null, how will this exception propagate? Will this be converted to an error that will make sense to the end user?  If it is more of an internal error, should we say so to the user.

Reply:
Yes, currently we are showing a custom message which is **Unable to get the current computer for this build.** subjected to @mw-hrastega  approval